### Yasin Guliyev

# Session 1 case study

### Import HR Data to Python.

### Step1.Import the necessary libraries.

In [1]:
import pandas as pd
import statistics as st

### Step 2. Call our Data as HR

In [2]:
data = pd.read_excel('HR Analytics.xlsx')

### Step 3. See the first 5 entries, then 12 entries.

In [3]:
data.head(5)

Age Attrition     BusinessTravel  DailyRate              Department  \
0   41       Yes      Travel_Rarely       1102                   Sales   
1   49        No  Travel_Frequently        279  Research & Development   
2   37       Yes      Travel_Rarely       1373  Research & Development   
3   33        No  Travel_Frequently       1392  Research & Development   
4   27        No      Travel_Rarely        591  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
0                 1          2  Life Sciences              1               1   
1                 8          1  Life Sciences              1               2   
2                 2          2          Other              1               4   
3                 3          4  Life Sciences              1               5   
4                 2          1        Medical              1               7   

   ...  RelationshipSatisfaction StandardHours  StockOptionLevel  \
0  ...                         1            80                 0   
1  ...                         4            80                 1   
2  ...                         2            80                 0   
3  ...                         3            80                 0   
4  ...                         4            80                 1   

   TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                  8                      0               1               6   
1                 10                      3               3              10   
2                  7                      3               3               0   
3                  8                      3               3               8   
4                  6                      3               3               2   

  YearsInCurrentRole  YearsSinceLastPromotion  YearsWithCurrManager  
0                  4                        0                     5  
1                  7                        1                     7  
2                  0                        0                     0  
3                  7                        3                     0  
4                  2                        2                     2  

[5 rows x 35 columns]

In [6]:
data.head(12)

Age Attrition     BusinessTravel  DailyRate              Department  \
0    41       Yes      Travel_Rarely       1102                   Sales   
1    49        No  Travel_Frequently        279  Research & Development   
2    37       Yes      Travel_Rarely       1373  Research & Development   
3    33        No  Travel_Frequently       1392  Research & Development   
4    27        No      Travel_Rarely        591  Research & Development   
5    32        No  Travel_Frequently       1005  Research & Development   
6    59        No      Travel_Rarely       1324  Research & Development   
7    30        No      Travel_Rarely       1358  Research & Development   
8    38        No  Travel_Frequently        216  Research & Development   
9    36        No      Travel_Rarely       1299  Research & Development   
10   35        No      Travel_Rarely        809  Research & Development   
11   29        No      Travel_Rarely        153  Research & Development   

    DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
0                  1          2  Life Sciences              1               1   
1                  8          1  Life Sciences              1               2   
2                  2          2          Other              1               4   
3                  3          4  Life Sciences              1               5   
4                  2          1        Medical              1               7   
5                  2          2  Life Sciences              1               8   
6                  3          3        Medical              1              10   
7                 24          1  Life Sciences              1              11   
8                 23          3  Life Sciences              1              12   
9                 27          3        Medical              1              13   
10                16          3        Medical              1              14   
11                15          2  Life Sciences              1              15   

    ...  RelationshipSatisfaction StandardHours  StockOptionLevel  \
0   ...                         1            80                 0   
1   ...                         4            80                 1   
2   ...                         2            80                 0   
3   ...                         3            80                 0   
4   ...                         4            80                 1   
5   ...                         3            80                 0   
6   ...                         1            80                 3   
7   ...                         2            80                 1   
8   ...                         2            80                 0   
9   ...                         2            80                 2   
10  ...                         3            80                 1   
11  ...                         4            80                 0   

    TotalWorkingYears  TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  \
0                   8                      0               1               6   
1                  10                      3               3              10   
2                   7                      3               3               0   
3                   8                      3               3               8   
4                   6                      3               3               2   
5                   8                      2               2               7   
6                  12                      3               2               1   
7                   1                      2               3               1   
8                  10                      2               3               9   
9                  17                      3               2               7   
10                  6                      5               3               5   
11                 10                      3               3               9   

   YearsInCurrentRole  YearsSinceLa

### Step 4. See the last 2, then 9 entries.

In [5]:
data.tail(2)

Age Attrition     BusinessTravel  DailyRate              Department  \
1468   49        No  Travel_Frequently       1023                   Sales   
1469   34        No      Travel_Rarely        628  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  \
1468                 2          3        Medical              1   
1469                 8          3        Medical              1   

      EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
1468            2065  ...                         4            80   
1469            2068  ...                         1            80   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
1468                 0                 17                      3   
1469                 0                  6                      3   

     WorkLifeBalance  YearsAtCompany YearsInCurrentRole  \
1468               2               9                  6   
1469               4               4                  3   

      YearsSinceLastPromotion  YearsWithCurrManager  
1468                        0                     8  
1469                        1                     2  

[2 rows x 35 columns]

In [7]:
data.tail(9)

Age Attrition     BusinessTravel  DailyRate              Department  \
1461   50       Yes      Travel_Rarely        410                   Sales   
1462   39        No      Travel_Rarely        722                   Sales   
1463   31        No         Non-Travel        325  Research & Development   
1464   26        No      Travel_Rarely       1167                   Sales   
1465   36        No  Travel_Frequently        884  Research & Development   
1466   39        No      Travel_Rarely        613  Research & Development   
1467   27        No      Travel_Rarely        155  Research & Development   
1468   49        No  Travel_Frequently       1023                   Sales   
1469   34        No      Travel_Rarely        628  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  \
1461                28          3      Marketing              1   
1462                24          1      Marketing              1   
1463                 5          3        Medical              1   
1464                 5          3          Other              1   
1465                23          2        Medical              1   
1466                 6          1        Medical              1   
1467                 4          3  Life Sciences              1   
1468                 2          3        Medical              1   
1469                 8          3        Medical              1   

      EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
1461            2055  ...                         2            80   
1462            2056  ...                         1            80   
1463            2057  ...                         2            80   
1464            2060  ...                         4            80   
1465            2061  ...                         3            80   
1466            2062  ...                         1            80   
1467            2064  ...                         2            80   
1468            2065  ...                         4            80   
1469            2068  ...                         1            80   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
1461                 1                 20                      3   
1462                 1                 21                      2   
1463                 0                 10                      2   
1464                 0                  5                      2   
1465                 1                 17                      3   
1466                 1                  9                      5   
1467                 1                  6                      0   
1468                 0                 17                      3   
1469                 0                  6                      3   

     WorkLifeBalance  YearsAtCompany YearsInCurrentRole  \
1461               3               3                  2   
1462               2              20                  9   
1463               3               9                  4   
1464               3               4                  2   
1465               3               5                  2   
1466               3               7                  7   
1467               3               6                  2   
1468               2               9                  6   
1469               4               4                  3   

      YearsSinceLastPromotion  YearsWithCurrManager  
1461                        2                     0  
1462                        9                     6  
1463                        1                     7  
1464                        0                     0  
1465                        0                     3  
1466                        1                     7  
1467                        0                     3  
1468                        0                     8  
1469                        1                     2  

[9 rows x 35 columns]

### Step 5. What is the number of observations in the dataset?

In [9]:
data.shape

(1470, 35)

In [10]:
1470*35

51450

### Step 6.Groupby age and return the mean of the remaining columns in each group.

In [16]:
data.groupby('Age').mean()

DailyRate  DistanceFromHome  Education  EmployeeCount  EmployeeNumber  \
Age                                                                           
18   747.625000          6.125000   2.500000            1.0     1053.625000   
19   548.444444         10.555556   2.111111            1.0      675.666667   
20   819.454545          7.090909   2.636364            1.0     1021.000000   
21   762.846154          9.076923   1.615385            1.0      951.538462   
22   806.937500          8.437500   1.812500            1.0      869.250000   
23   707.928571          9.142857   1.857143            1.0     1225.500000   
24   863.115385         11.884615   2.115385            1.0     1011.384615   
25   768.076923          8.769231   1.923077            1.0     1005.307692   
26   844.769231         10.230769   2.641026            1.0      981.051282   
27   907.333333          7.500000   2.666667            1.0     1001.958333   
28   927.500000          8.875000   2.791667            1.0     1120.125000   
29   779.308824         10.808824   2.573529            1.0     1100.867647   
30   791.266667          9.850000   2.766667            1.0      866.766667   
31   725.173913          9.260870   3.057971            1.0     1070.000000   
32   766.262295          7.278689   3.049180            1.0      986.196721   
33   739.672414          8.793103   3.086207            1.0     1009.120690   
34   788.285714          8.909091   3.090909            1.0     1080.987013   
35   825.320513         11.333333   3.217949            1.0     1055.551282   
36   819.130435          8.246377   3.246377            1.0     1045.971014   
37   827.720000          8.480000   3.000000            1.0      815.280000   
38   776.155172          7.724138   2.965517            1.0     1058.724138   
39   774.642857          9.333333   2.785714            1.0     1224.880952   
40   841.719298          9.385965   2.964912            1.0     1142.508772   
41   721.525000          8.050000   2.925000            1.0      983.650000   
42   713.391304         10.782609   2.913043            1.0     1051.434783   
43   818.250000         11.406250   3.125000            1.0      998.812500   
44   856.848485          9.030303   3.181818            1.0     1132.757576   
45   844.975610         11.097561   2.975610            1.0     1008.512195   
46   802.242424          6.545455   2.878788            1.0     1021.939394   
47   852.875000          9.125000   3.125000            1.0     1229.291667   
48   816.263158          8.105263   3.157895            1.0     1168.421053   
49   886.541667         10.083333   2.958333            1.0     1142.041667   
50   879.100000          9.833333   3.166667            1.0      883.166667   
51   953.789474          7.684211   3.263158            1.0      692.368421   
52   796.666667          9.111111   3.222222            1.0      865.722222   
53   856.684211          7.631579   3.052632            1.0      918.157895   
54   727.666667         10.277778   3.500000            1.0     1032.666667   
55   646.227273          8.909091   3.000000            1.0     1042.772727   
56   904.428571          9.428571   3.214286            1.0     1151.571429   
57   385.500000         13.750000   2.750000            1.0      689.000000   
58   690.500000          8.785714   3.357143            1.0     1020.142857   
59   992.300000          6.400000   2.800000            1.0      556.900000   
60   833.200000         11.800000   3.600000            1.0      956.800000   

     EnvironmentSatisfaction  HourlyRate  JobInvolvement  JobLevel  \
Age                                                                  
18                  2.750000   68.125000        2.875000  1.000000   
19                  2.444444   63.555556        2.444444  1.000000   
20                  3.181818   61.909091        2.636364  1.000000   
21                  2.692308   67.307692        2.692308  1.000000   
22                  2.812500  

### Step7. Check if it has null values , then show sum of them

In [17]:
data.isnull()

Age  Attrition  BusinessTravel  DailyRate  Department  \
0     False      False           False      False       False   
1     False      False           False      False       False   
2     False      False           False      False       False   
3     False      False           False      False       False   
4     False      False           False      False       False   
...     ...        ...             ...        ...         ...   
1465  False      False           False      False       False   
1466  False      False           False      False       False   
1467  False      False           False      False       False   
1468  False      False           False      False       False   
1469  False      False           False      False       False   

      DistanceFromHome  Education  EducationField  EmployeeCount  \
0                False      False           False          False   
1                False      False           False          False   
2                False      False           False          False   
3                False      False           False          False   
4                False      False           False          False   
...                ...        ...             ...            ...   
1465             False      False           False          False   
1466             False      False           False          False   
1467             False      False           False          False   
1468             False      False           False          False   
1469             False      False           False          False   

      EmployeeNumber  ...  RelationshipSatisfaction  StandardHours  \
0              False  ...                     False          False   
1              False  ...                     False          False   
2              False  ...                     False          False   
3              False  ...                     False          False   
4              False  ...                     False          False   
...              ...  ...                       ...            ...   
1465           False  ...                     False          False   
1466           False  ...                     False          False   
1467           False  ...                     False          False   
1468           False  ...                     False          False   
1469           False  ...                     False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
1465             False              False                  False   
1466             False              False                  False   
1467             False              False                  False   
1468             False              False                  False   
1469             False              False                  False   

      WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0               False           False               False   
1               False           False               False   
2               False           False               False   
3               False           False               False   
4               False           False               False   
...               ...             ...                 ...   
1465            False           False               False   
1466            False           False               False   
1467            False           False               False   
1468            False           False               False   
1469            False           False               False   

      YearsSinceLas

In [18]:
data.isnull().sum()

Age                         0
Attrition                   0
BusinessTravel              0
DailyRate                   0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeNumber              0
EnvironmentSatisfaction     0
Gender                      0
HourlyRate                  0
JobInvolvement              0
JobLevel                    0
JobRole                     0
JobSatisfaction             0
MaritalStatus               0
MonthlyIncome               0
MonthlyRate                 0
NumCompaniesWorked          0
Over18                      0
OverTime                    0
PercentSalaryHike           0
PerformanceRating           0
RelationshipSatisfaction    0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           0
TrainingTimesLastYear       0
WorkLifeBalance             0
YearsAtCompany              0
YearsInCurrentRole          0
YearsSince

### Step 8. What is the mean of Hourly Rate?

In [20]:
data['HourlyRate'].mean()

65.89115646258503

### Step 9. What is max&min age of employees?

In [21]:
data['Age'].max()

60

In [22]:
data['Age'].min()

18

### Step 10. What is the most frequent level of Education that  employees have?

In [23]:
data['Education'].mode()

0    3
dtype: int64

### Step 11. What is median of total working years ?

In [25]:
data['TotalWorkingYears'].median()

10.0

### Step 12. Sort values by YearsatCompany in ascending order.

In [27]:
data.sort_values('YearsAtCompany', ascending = True)

Age Attrition     BusinessTravel  DailyRate              Department  \
1311   18        No         Non-Travel       1431  Research & Development   
1371   56        No      Travel_Rarely       1443                   Sales   
460    26        No      Travel_Rarely        775                   Sales   
1068   28       Yes  Travel_Frequently        289  Research & Development   
457    18       Yes  Travel_Frequently       1306                   Sales   
...   ...       ...                ...        ...                     ...   
561    52        No      Travel_Rarely        621                   Sales   
1116   55        No      Travel_Rarely        685                   Sales   
270    55        No      Travel_Rarely        452  Research & Development   
98     58        No      Travel_Rarely        682                   Sales   
126    58       Yes      Travel_Rarely        147  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  \
1311                14          3        Medical              1   
1371                11          5      Marketing              1   
460                 29          2        Medical              1   
1068                 2          2        Medical              1   
457                  5          3      Marketing              1   
...                ...        ...            ...            ...   
561                  3          4      Marketing              1   
1116                26          5      Marketing              1   
270                  1          3        Medical              1   
98                  10          4        Medical              1   
126                 23          4        Medical              1   

      EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
1311            1839  ...                         3            80   
1371            1935  ...                         3            80   
460              618  ...                         1            80   
1068            1504  ...                         3            80   
457              614  ...                         4            80   
...              ...  ...                       ...           ...   
561              776  ...                         1            80   
1116            1578  ...                         3            80   
270              374  ...                         3            80   
98               131  ...                         3            80   
126              165  ...                         4            80   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
1311                 0                  0                      4   
1371                 1                  6                      3   
460                  2                  8                      5   
1068                 0                  8                      2   
457                  0                  0                      3   
...                ...                ...                    ...   
561                  0                 34                      3   
1116                 1                 36                      3   
270                  0                 37                      2   
98                   0                 38                      1   
126                  1                 40                      3   

     WorkLifeBalance  YearsAtCompany YearsInCurrentRole  \
1311               1               0                  0   
1371               3               0                  0   
460                3               0                  0   
1068               2               0                  0   
457                3               0                  0   
...              ...             ...                ...   
561                4              34                  6   
1116               3              36                  6   
270                3              36                 10   
98                 2              37    

### Step 13. What is difference between max and min age of employees?

In [28]:
data['Age'].max()-data['Age'].min()

42

### Step 14. What is Standard deviation of the Age of Employees?

In [29]:
data['Age'].std()

9.135373489136734

### Step 15. Get a statistical summary of the data set. 
### Give Interpretation for it

### I selected the column which named as DailyRate
### In this column the mean is 802.4857142857143
### The mode is 0 and 691
### The median is 802.0
### The standart deviation of this column is 403.50909994352827

### Step 16.Summarize only the DistanceFromHome.

In [32]:
data['DistanceFromHome'].describe()

count    1470.000000
mean        9.192517
std         8.106864
min         1.000000
25%         2.000000
50%         7.000000
75%        14.000000
max        29.000000
Name: DistanceFromHome, dtype: float64

### Step17.Check its info

In [30]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 35 columns):
Age                         1470 non-null int64
Attrition                   1470 non-null object
BusinessTravel              1470 non-null object
DailyRate                   1470 non-null int64
Department                  1470 non-null object
DistanceFromHome            1470 non-null int64
Education                   1470 non-null int64
EducationField              1470 non-null object
EmployeeCount               1470 non-null int64
EmployeeNumber              1470 non-null int64
EnvironmentSatisfaction     1470 non-null int64
Gender                      1470 non-null object
HourlyRate                  1470 non-null int64
JobInvolvement              1470 non-null int64
JobLevel                    1470 non-null int64
JobRole                     1470 non-null object
JobSatisfaction             1470 non-null int64
MaritalStatus               1470 non-null object
MonthlyIncome         

### Step 18.What is the most frequent level of education that employees have?

In [33]:
data['Education'].mode()

0    3
dtype: int64

### Step19.Display our dataframe in which Education which are more than 3.00

In [34]:
data[data['Education'] > 3.00]

Age Attrition     BusinessTravel  DailyRate              Department  \
3      33        No  Travel_Frequently       1392  Research & Development   
15     29        No      Travel_Rarely       1389  Research & Development   
18     53        No      Travel_Rarely       1219                   Sales   
21     36       Yes      Travel_Rarely       1218                   Sales   
22     34        No      Travel_Rarely        419  Research & Development   
...   ...       ...                ...        ...                     ...   
1455   40        No      Travel_Rarely       1322  Research & Development   
1456   35        No  Travel_Frequently       1199  Research & Development   
1457   40        No      Travel_Rarely       1194  Research & Development   
1458   35        No      Travel_Rarely        287  Research & Development   
1460   29        No      Travel_Rarely        468  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  \
3                    3          4  Life Sciences              1   
15                  21          4  Life Sciences              1   
18                   2          4  Life Sciences              1   
21                   9          4  Life Sciences              1   
22                   7          4  Life Sciences              1   
...                ...        ...            ...            ...   
1455                 2          4  Life Sciences              1   
1456                18          4  Life Sciences              1   
1457                 2          4        Medical              1   
1458                 1          4  Life Sciences              1   
1460                28          4        Medical              1   

      EmployeeNumber  ...  RelationshipSatisfaction StandardHours  \
3                  5  ...                         3            80   
15                20  ...                         3            80   
18                23  ...                         3            80   
21                27  ...                         2            80   
22                28  ...                         3            80   
...              ...  ...                       ...           ...   
1455            2048  ...                         4            80   
1456            2049  ...                         4            80   
1457            2051  ...                         2            80   
1458            2052  ...                         4            80   
1460            2054  ...                         2            80   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
3                    0                  8                      3   
15                   1                 10                      1   
18                   0                 31                      3   
21                   0                 10                      4   
22                   0                 13                      4   
...                ...                ...                    ...   
1455                 0                  8                      2   
1456                 2                 10                      2   
1457                 3                 20                      2   
1458                 1                  4                      5   
1460                 0                  5                      3   

     WorkLifeBalance  YearsAtCompany YearsInCurrentRole  \
3                  3               8                  7   
15                 3              10                  9   
18                 3              25                  8   
21                 3               5                  3   
22                 3              12                  6   
...              ...             ...                ...   
1455               3               2                  2   
1456               4              10                  2   
1457               3               5                  3   
1458               3               4    